azure-pipelines.yml

```diff
@@ -196,16 +201,26 @@ jobs:
     inputs:
       targetType: inline
       script: |
-        $InstallArgs = @(
-          "/I"
-          '"$env:SYSTEM_ARTIFACTSDIRECTORY/msi/Microsoft Azure CLI.msi"'
-          "/norestart"
-          "/L*v"
-          ".\install_logs.txt"
-        )
-        Start-Process "msiexec.exe" -ArgumentList $InstallArgs -Wait -NoNewWindow
+        # Elevate as Admin
+        if (-NOT ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")) {  
+          $arguments = "& '" +$myinvocation.mycommand.definition + "'"
+          Start-Process powershell -Verb runAs -ArgumentList $arguments
+          Break
+        }
+        az --version
+        $pre_installed_version=az version --query '\"azure-cli\"' -o tsv
+        $to_be_installed_version=Get-Content $(System.ArtifactsDirectory)/metadata/version
+        if ($pre_installed_version -eq $to_be_installed_version){
+          $reinstall_option="REINSTALL=ALL REINSTALLMODE=emus"
+        }else {
+          $reinstall_option=""
+        }
+        Start-Process "msiexec.exe" -ArgumentList "/i `"$env:SYSTEM_ARTIFACTSDIRECTORY\msi\Microsoft Azure CLI.msi`" $reinstall_option /q /norestart /l*v .\install_logs.txt" -Wait -NoNewWindow
```

Comment:
`$InstallArgs` looks more readable to me.

Reply:
Reverted